### PR TITLE
Start removing some of the old v1 FS functions

### PR DESCRIPTION
**Pull request by curiousdannii**
Started working on #12185 

I removed the functions that weren't being used anywhere else (leaving abort calls in ASSERTIONS), and also turned one function that was being used elsewhere, `mmapAlloc`, into a top level function. Does that seem like a good strategy for the other functions which are more commonly used in the code? And are functions like `createPreloadedFile` actually old?

### CODE REVIEW

**Review by sbc100 — APPROVED** [review]
This looks totally great to me.   Thanks for working on it!

I'll wait for an lgtm from @juj or @kripken before landing...

File: src/library_fs.js
Comment (kripken) [inline src/library_fs.js:2008]: How about mentioning the replacement in these error messages? That would also help confirm in review that we are only removing things with a more modern variant.

**Review by kripken — COMMENTED** [review]
In general this sounds great, but let's make sure to document the better replacements, see comment below - without that, I'm actually unsure if these are redundant or not.

Moving more code to top-level functions is a good idea, yes!

File: src/library_fs.js
Comment (curiousdannii) [reply src/library_fs.js:2008]: Okay, for some like `createFolder` I can suggest `mkdir`. `absolutePath` was not used, tested, or documented anywhere in the project, so I thought we could just remove it, but considering it's just a one-line wrapper around PATH_FS I can add an explanation for it too. What about for `joinPath`? Still just suggest `PATH.join` even though it does more than just wrap it?

Comment (curiousdannii) [reply src/library_fs.js]: Okay, I've done that. People will have to account for the different functionality of the newer functions, but that's what they have to do anytime something is deprecated.

**Comment by curiousdannii** [comment]
I'm getting an es-check error on test_fs_nodefs_rw, but when I try it locally it says it's fine.

**Comment by sbc100** [comment]
Are you running the wasm0.xxxx version of the test locally?

**Comment by sbc100** [comment]
I can't see how this change could generate the es-check failure.. odd.

**Comment by curiousdannii** [comment]
I just manually compiled it with the arguments from the test.

**Review by kripken — APPROVED** [review]

**Comment by kripken** [comment]
I'm also not sure what could cause that error. Maybe after resolving the merge conflict it'll go away, if it was something old on trunk?

**Comment by curiousdannii** [comment]
Well I'm not sure what changed, but after rebasing the tests all passed :)

**Pull request merged**
